views/js/checkout-component-renderer.js

```diff
@@ -492,17 +497,49 @@ jQuery(document).ready(function() {
         }
 
         function handleOnClick(resolve, reject) {
-            const paymentMethodContainer = this.paymentForm.closest('.adyen-payment');
+            const paymentMethodContainer = getFormPaymentMethodContainer(this.paymentForm);
+            const paymentMethodType = paymentMethodContainer.data('local-payment-method');
             // Show message if button is disabled else if not in progress, hide and resolve
             if (prestaShopPlaceOrderButton.prop('disabled') && !isPlaceOrderInProgress()) {
                 showRequiredConditionsInfoMessage(paymentMethodContainer);
-                reject(new Error('Terms of service not agreed'));
+                if (paymentMethodType === 'paypal') {
+                    return false;
+                } else {
+                    reject(new Error('Terms of service not agreed'));
+                }
```

Comment:
If I understand correctly Paypal won't support T&Cs? If that's the case let's make a mention about it in the PR description so that it could be added to the docs upon release @Morerice

Reply:
Actually there is a bug in the current version fo the component and this is the workaround to support the onClick callback. But it does the same like the other components we just need to return false instead of reject()